src/js/stores/search.js

```diff
@@ -259,17 +264,23 @@ define(function (require, exports, module) {
         },
 
         /**
-         * Get subcategories for all the headers
+         * Get filters that should be displayed 
+         * Don't include filters that don't have items or that aren't valid for the search type
          *
-         * @param {Array.<string>} headers
-         * @return {Array.<Immutable.List>}
+         * @param {Immutable.List} items
+         * @param {string} type For looking up the search
+         * @return {Array.<string>}
          */
-        _getFiltersByHeaders: function (headers) {
-            return _.map(headers, function (header) {
-                if (this._registeredSearchTypes[header]) {
-                    return this._registeredSearchTypes[header].filters;
-                }
-            }.bind(this), []);
+        _getFiltersByItems: function (items, type) {
+            if (items.size === 0) {
+                return [];
+            }
+            var moduleFilters = this._registeredSearchTypes[type].filters.toJS(),
+                itemCategories = collection.pluck(items, "category").toJS(),
```

Comment:
`flatten()` isn't flattening it, I think because the category lists are just arrays, not Immutable.

But I can make it more readable using `_.flatten()`